wallet-new/src/Cardano/Wallet/Kernel/Pending.hs

```diff
@@ -108,24 +108,26 @@ newTx ActiveWallet{..} accountId tx partialMeta upd = do
             putTxMeta (walletPassive ^. walletMeta) txMeta
             submitTx
             return (Right txMeta)
-    where
-        (txOut :: [TxOut]) = NE.toList $ (_txOutputs . taTx $ tx)
-
-        -- | NOTE: we recognise addresses in the transaction outputs that belong to _all_ wallets,
-        --  not only for the wallet to which this transaction is being submitted
-        allOurs :: [(WalletId, EncryptedSecretKey)] -> [(HdAddress,Coin)]
-        allOurs = concatMap ourAddrs
-
-        ourAddrs :: (WalletId, EncryptedSecretKey) -> [(HdAddress,Coin)]
-        ourAddrs (wid, esk) =
-            map f $ filterOurs wKey txOutAddress txOut
-            where
-                f (txOut',addressId) = (initHdAddress addressId (txOutAddress txOut'), txOutValue txOut')
-                wKey = (wid, keyToWalletDecrCredentials $ KeyForRegular esk)
-
-        submitTx :: IO ()
-        submitTx = modifyMVar_ (walletPassive ^. walletSubmission) $
-                    return . addPending accountId (Pending.singleton tx)
+  where
+    (txOut :: [TxOut]) = NE.toList $ (_txOutputs . taTx $ tx)
+
+    -- | NOTE: we recognise addresses in the transaction outputs that belong to _all_ wallets,
+    --  not only for the wallet to which this transaction is being submitted
+    allOurs :: [(WalletId, WalletUserKey)] -> [(HdAddress,Coin)]
+    allOurs = concatMap ourAddrs
+
+    ourAddrs :: (WalletId, WalletUserKey) -> [(HdAddress,Coin)]
+    ourAddrs (wid, walletUserKey) =
+        map f $ filterOurs wKey txOutAddress txOut
+      where
+        f (txOut',addressId) = (initHdAddress addressId (txOutAddress txOut'), txOutValue txOut')
+        wKey = case walletUserKey of
+                   RegularWalletKey esk -> (wid, keyToWalletDecrCredentials $ KeyForRegular esk)
+                   ExternalWalletKey pk -> (wid, keyToWalletDecrCredentials $ KeyForExternal pk)
+
+    submitTx :: IO ()
+    submitTx = modifyMVar_ (walletPassive ^. walletSubmission) $
+                return . addPending accountId (Pending.singleton tx)
```

Comment:
What's happening here :/ ? May we avoid re-indentation of the whole block to make the changes obvious (EncryptedSecretKey -> WalletUserKey I believe)